crates/miden-lib/asm/kernels/transaction/lib/epilogue.masm

```diff
@@ -286,7 +307,18 @@ export.finalize_transaction
 
     # compute the fee the tx needs to pay
     # ignored for now
-    exec.compute_fee drop
+    exec.compute_fee
+    # => [fee_amount]
+
+    exec.build_native_fee_asset
+    # => [FEE_ASSET]
+
+    # remove the fee from the native account's vault
+    exec.account::remove_asset_from_vault
+    # => [FEE_ASSET]
```

Comment:
Probably not for this PR, but should we check if the vault has the asset? This check will be performed in `account::remove_asset_from_vault` and if not, we'll panic - but I think the error message would be pretty generic. It would be nice to return a more tailored error - e.g., "Insufficient fee". It would incur the cost of extra access to the vault - but maybe that's an acceptable tradeoff?